receiver/k8sobjectsreceiver/receiver_test.go

```diff
@@ -136,5 +137,71 @@ func TestWatchObject(t *testing.T) {
 	assert.Len(t, consumer.Logs(), 2)
 	assert.Equal(t, 2, consumer.Count())
 
+	mockClient.deletePods(
+		generatePod("pod2", "default", map[string]interface{}{
+			"environment": "test",
+		}, "2"),
+	)
+	time.Sleep(time.Millisecond * 100)
+	assert.Len(t, consumer.Logs(), 3)
+	assert.Equal(t, 3, consumer.Count())
+
+	assert.NoError(t, r.Shutdown(ctx))
+}
+
+func TestExludeDeletedTrue(t *testing.T) {
+	t.Parallel()
+
+	mockClient := newMockDynamicClient()
+
+	mockClient.createPods(
+		generatePod("pod1", "default", map[string]interface{}{
+			"environment": "production",
+		}, "1"),
+	)
+
+	rCfg := createDefaultConfig().(*Config)
+	rCfg.makeDynamicClient = mockClient.getMockDynamicClient
+	rCfg.makeDiscoveryClient = getMockDiscoveryClient
+
+	rCfg.Objects = []*K8sObjectsConfig{
+		{
+			Name:       "pods",
+			Mode:       WatchMode,
+			Namespaces: []string{"default"},
+			ExcludeWatchType: []apiWatch.EventType{
+				apiWatch.Deleted,
+			},
+		},
+	}
+
+	err := rCfg.Validate()
+	require.NoError(t, err)
+
+	consumer := newMockLogConsumer()
+	r, err := newReceiver(
+		receivertest.NewNopCreateSettings(),
+		rCfg,
+		consumer,
+	)
+
+	ctx := context.Background()
+	require.NoError(t, err)
+	require.NotNil(t, r)
+	require.NoError(t, r.Start(ctx, componenttest.NewNopHost()))
+
+	time.Sleep(time.Millisecond * 100)
```

Comment:
Perhaps an `assert.Never` would be better here?

Reply:
The unit tests definitely need some work, but I'd prefer to do it in a separate PR.